transcendence-app/src/chat/chat.controller.ts

```diff
@@ -95,6 +96,28 @@ export class ChatController {
     return ret;
   }
 
+  @Delete('room/:chatroomId/members')
+  @ApiOkResponse({
+    description: 'Authenticated user leaves the given chatroom',
+    type: ChatroomMember,
+  })
+  @ApiBadRequestResponse({ description: 'Bad Request' })
+  @ApiNotFoundResponse({ description: 'Not Found' })
+  @ApiServiceUnavailableResponse({ description: 'Service Unavailable' })
+  async leaveChatroom(
+    @GetUser() user: User,
+    @Param('chatroomId', ParseUUIDPipe) chatroomId: string,
+  ): Promise<ChatroomMember> {
+    const chatroomMember = await this.chatroomMemberService.leaveChatroom(
```

Comment:
I think we should prevent this request from succeeding if the auth user is the owner.
I would say an owner can remove a chatroom, but not leave it, since we are not making ownership transferable.

Reply:
(I just tried, and it's possible. The way it is, we can have this weird situation where the owner of a chatroom is not a part of that chatroom).

Reply:
Good point :+1:, but I think that may be fixed in another task. It will imply having all that logic of transfering ownership (new service method, new UI to choose a successor etc.)

So, for me, not strictly blocking if we are aware of this in a new issue.

Reply:
Discussing it in slack.
Non blocking.

Reply:
> It will imply having all that logic of transfering ownership (new service method, new UI to choose a successor etc.)

Apparently we don't care about that. [Here](https://42born2code.slack.com/archives/C033YQ28QA0/p1668076594354449?thread_ts=1668072399.764979&cid=C033YQ28QA0)

Reply:
Updated, as discussed on Slack (we delete the chatroom when the owner leaves).
Thanks for the suggestions.